[Automation] native-image run fails for io.netty:netty-codec-dns:4.1.79.Final

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-dns:4.1.74.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25251908980/job/74045252761
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
-----------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (37.8s @ 0.87GB)
    8,552 types,   9,840 fields, and  38,509 methods found reachable
    2,901 types,     285 fields, and   2,453 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (4.4s @ 0.96GB)
[4/8] Parsing methods...      [**]                                                                       (4.4s @ 0.64GB)
[5/8] Inlining methods...     [***]                                                                      (2.5s @ 0.70GB)
[6/8] Compiling methods...    [****]                                                                    (20.2s @ 0.82GB)
[7/8] Laying out methods...   [**]                                                                       (3.6s @ 0.77GB)
[8/8] Creating image...       [**]                                                                       (3.7s @ 0.90GB)
   9.88MB (35.49%) for code area:    23,871 compilation units
  14.02MB (50.38%) for image heap:  214,768 objects and 155 resources
   3.93MB (14.13%) for other data
  27.84MB in total image size, 25.69MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   4.55MB java.base                                            2.90MB byte[] for code metadata
   2.14MB svm.jar (Native Image)                               2.63MB byte[] for java.lang.String
 560.94kB assertj-core-3.22.0.jar                              1.39MB java.lang.String
 481.09kB java.xml                                             1.08MB java.lang.Class
 268.96kB netty-common-4.1.79.Final.jar                      912.58kB byte[] for general heap data
 214.92kB netty-buffer-4.1.79.Final.jar                      889.41kB com.oracle.svm.core.hub.DynamicHubCompanion
 198.00kB junit-jupiter-engine-5.9.2.jar                     564.58kB heap alignment
 120.80kB junit-platform-launcher-1.9.2.jar                  450.30kB java.util.concurrent.ConcurrentHashMap$Node
 103.55kB java.logging                                       300.61kB java.util.HashMap$Node
  99.49kB junit-platform-engine-1.9.2.jar                    281.70kB java.lang.Object[]
 768.35kB for 29 more packages                                 2.64MB for 2033 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 22 component(s) is embedded in binary (1.07kB). 1 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        5.2s (6.0% of total time) in 742 GCs | Peak RSS: 1.72GB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-dns_tests-tests (executable)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'io.netty.netty-codec-dns_tests-tests' in 1m 25s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::getLong has been called by io.netty.util.internal.PlatformDependent0$4 (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-dns_tests-tests)
WARNING: Please consider reporting this to the maintainers of class io.netty.util.internal.PlatformDependent0$4
WARNING: sun.misc.Unsafe::getLong will be removed in a future release
io_netty.netty_codec_dns.Netty_codec_dnsTest > dnsMessagesManageSectionsFlagsAndReferenceCountedRecords() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > datagramQueryEncoderAndDecoderRoundTripEnvelopeHeaderAndSections() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > registryTypesCodesAndOpcodesResolveKnownAndUnknownValues() SUCCESSFUL

io_netty.netty_codec_dns.Netty_codec_dnsTest > recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords() FAILED


Failures (6):
  JUnit Jupiter:Netty_codec_dnsTest:dnsMessagesManageSectionsFlagsAndReferenceCountedRecords()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'dnsMessagesManageSectionsFlagsAndReferenceCountedRecords', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       io.netty.buffer.Unpooled.<clinit>(Unpooled.java:75)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.ipv4Record(Netty_codec_dnsTest.java:358)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.dnsMessagesManageSectionsFlagsAndReferenceCountedRecords(Netty_codec_dnsTest.java:54)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
     Caused by: java.lang.IllegalArgumentException: Can't find '[toLeakAwareBuffer]' in io.netty.buffer.AbstractByteBufAllocator
       io.netty.util.ResourceLeakDetector.addExclusions(ResourceLeakDetector.java:596)
       io.netty.buffer.AbstractByteBufAllocator.<clinit>(AbstractByteBufAllocator.java:37)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_dns.Netty_codec_dnsTest.recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData(Netty_codec_dnsTest.java:164)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.3/java.lang.Thread.runWith(Thread.java:1487)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_dns.Netty_codec_dnsTest.ipv4Record(Netty_codec_dnsTest.java:358)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers(Netty_codec_dnsTest.java:290)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:datagramQueryEncoderAndDecoderRoundTripEnvelopeHeaderAndSections()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'datagramQueryEncoderAndDecoderRoundTripEnvelopeHeaderAndSections', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.AbstractByteBufAllocator
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       io.netty.buffer.ByteBufUtil.<clinit>(ByteBufUtil.java:86)
       io.netty.buffer.ByteBufAllocator.<clinit>(ByteBufAllocator.java:24)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag(Netty_codec_dnsTest.java:249)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_dns.Netty_codec_dnsTest.recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords(Netty_codec_dnsTest.java:123)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.3/java.lang.Thread.runWith(Thread.java:1487)
       [...]

Test run finished after 7 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         7 tests found           ]
[         0 tests skipped         ]
[         7 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         6 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-dns_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 29s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.netty:netty-codec-dns:4.1.74.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 31s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.79.Final failed with exit code 1.
FAILED[native-image run][4.1.79.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-dns:4.1.74.Final"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-dns:4.1.74.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25251908980/job/74045252763
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
sion: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (30.3s @ 0.85GiB)
    8,636 types,   9,872 fields, and  38,926 methods found reachable
      314 types,     111 fields, and     723 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      155 resource accesses registered with 462.58KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (3.8s @ 0.98GiB)
[4/8] Parsing methods...      [**]                                                                      (3.4s @ 1.02GiB)
[5/8] Inlining methods...     [***]                                                                     (2.5s @ 0.62GiB)
[6/8] Compiling methods...    [****]                                                                   (17.6s @ 0.76GiB)
[7/8] Laying out methods...   [**]                                                                      (3.8s @ 0.78GiB)
[8/8] Creating image...       [**]                                                                      (3.4s @ 0.90GiB)
   8.75MiB (33.72%) for code area:    24,134 compilation units
  13.38MiB (51.55%) for image heap:  219,845 objects and 25 resources
   3.82MiB (14.73%) for other data
  25.94MiB in total image size, 23.88MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
 991.51KiB o.g.n.runtime/com.oracle.svm.core.code              2.72MiB byte[] for code metadata
 704.06KiB java.base/java.util                                 2.60MiB byte[] for string data
 408.45KiB java.base/jdk.internal.classfile.impl               1.39MiB java.base/java.lang.String
 322.84KiB java.base/java.lang                                 1.04MiB java.base/java.lang.Class
 244.43KiB java.base/java.util.concurrent                    944.56KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 228.99KiB java.base/java.text                               576.38KiB byte[] for general heap data
 224.55KiB java.base/com.sun.crypto.provider                 462.98KiB byte[] for embedded resources
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       427.85KiB byte[] for reflection metadata
 201.58KiB io.netty.buffer                                   334.92KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 152.36KiB java.base/java.util.stream                        299.06KiB java.base/java.util.HashMap$Node
   5.13MiB for 324 more packages                               2.65MiB for 2051 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 22 component(s) is embedded in binary (1.12KiB). 1 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.6s (7.5% of total time) in 695 GCs | Peak RSS: 1.65GiB | CPU load: 3.59
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-dns_tests-tests (executable, 23.88MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'io.netty.netty-codec-dns_tests-tests' in 1m 13s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

io_netty.netty_codec_dns.Netty_codec_dnsTest > dnsMessagesManageSectionsFlagsAndReferenceCountedRecords() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > datagramQueryEncoderAndDecoderRoundTripEnvelopeHeaderAndSections() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > registryTypesCodesAndOpcodesResolveKnownAndUnknownValues() SUCCESSFUL

io_netty.netty_codec_dns.Netty_codec_dnsTest > recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords() FAILED


Failures (6):
  JUnit Jupiter:Netty_codec_dnsTest:dnsMessagesManageSectionsFlagsAndReferenceCountedRecords()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'dnsMessagesManageSectionsFlagsAndReferenceCountedRecords', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       io.netty.buffer.Unpooled.<clinit>(Unpooled.java:75)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.ipv4Record(Netty_codec_dnsTest.java:358)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.dnsMessagesManageSectionsFlagsAndReferenceCountedRecords(Netty_codec_dnsTest.java:54)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
     Caused by: java.lang.IllegalArgumentException: Can't find '[toLeakAwareBuffer]' in io.netty.buffer.AbstractByteBufAllocator
       io.netty.util.ResourceLeakDetector.addExclusions(ResourceLeakDetector.java:596)
       io.netty.buffer.AbstractByteBufAllocator.<clinit>(AbstractByteBufAllocator.java:37)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_dns.Netty_codec_dnsTest.recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData(Netty_codec_dnsTest.java:164)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
       [...]
WARNING: sun.misc.Unsafe::getLong has been called by io.netty.util.internal.PlatformDependent0$4 (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-dns_tests-tests)
WARNING: Please consider reporting this to the maintainers of class io.netty.util.internal.PlatformDependent0$4
WARNING: sun.misc.Unsafe::getLong will be removed in a future release

FAILURE: Build failed with an exception.

* What went wrong:
  JUnit Jupiter:Netty_codec_dnsTest:tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_dns.Netty_codec_dnsTest.ipv4Record(Netty_codec_dnsTest.java:358)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers(Netty_codec_dnsTest.java:290)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:datagramQueryEncoderAndDecoderRoundTripEnvelopeHeaderAndSections()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'datagramQueryEncoderAndDecoderRoundTripEnvelopeHeaderAndSections', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.AbstractByteBufAllocator
       io.netty.buffer.ByteBufUtil.<clinit>(ByteBufUtil.java:86)
       io.netty.buffer.ByteBufAllocator.<clinit>(ByteBufAllocator.java:24)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.datagramQueryEncoderAndDecoderRoundTripEnvelopeHeaderAndSections(Netty_codec_dnsTest.java:209)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-dns_tests-tests'' finished with non-zero exit value 1
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag(Netty_codec_dnsTest.java:249)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled

* Try:
       io_netty.netty_codec_dns.Netty_codec_dnsTest.recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords(Netty_codec_dnsTest.java:123)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 17s
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
       [...]

Test run finished after 6 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         7 tests found           ]
[         0 tests skipped         ]
[         7 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         6 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.netty:netty-codec-dns:4.1.74.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 19s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.79.Final failed with exit code 1.
FAILED[native-image run][4.1.79.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-dns:4.1.74.Final"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-dns:4.1.74.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25251908980/job/74045252770
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
lder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 1 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (34.2s @ 0.95GiB)
    9,176 types,  10,674 fields, and  41,958 methods found reachable
      442 types,     111 fields, and     909 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      157 resource accesses registered with 462.72KiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.2s @ 1.05GiB)
[4/8] Parsing methods...      [**]                                                                      (4.6s @ 0.78GiB)
[5/8] Inlining methods...     [***]                                                                     (1.7s @ 0.85GiB)
[6/8] Compiling methods...    [****]                                                                   (20.4s @ 0.86GiB)
[7/8] Laying out methods...   [**]                                                                      (3.4s @ 1.04GiB)
[8/8] Creating image...       [**]                                                                      (3.9s @ 1.19GiB)
   9.87MiB (33.25%) for code area:    26,446 compilation units
  15.69MiB (52.85%) for image heap:  258,570 objects and 27 resources
   4.12MiB (13.89%) for other data
  29.68MiB in total image size, 27.44MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.10MiB o.g.n.runtime/com.oracle.svm.core.code              3.08MiB byte[] for code metadata
 711.19KiB java.base/java.util                                 2.89MiB byte[] for string data
 408.42KiB java.base/jdk.internal.classfile.impl               1.53MiB java.base/java.lang.String
 332.47KiB java.base/java.lang                                 1.11MiB java.base/java.lang.Class
 267.01KiB java.base/com.sun.crypto.provider                   0.98MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 244.56KiB java.base/java.util.concurrent                    924.95KiB byte[] for general heap data
 231.56KiB java.base/java.text                               495.97KiB byte[] for reflection metadata
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       463.16KiB byte[] for embedded resources
 201.58KiB io.netty.buffer                                   362.58KiB java.base/java.lang.Object[]
 194.33KiB java.base/sun.security.provider                   359.30KiB heap alignment
   6.03MiB for 334 more packages                               3.56MiB for 2204 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 22 component(s) is embedded in binary (1.12KiB). 1 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       6.5s (7.9% of total time) in 787 GCs | Peak RSS: 1.83GiB | CPU load: 3.60
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-dns_tests-tests (executable, 27.44MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'io.netty.netty-codec-dns_tests-tests' in 1m 21s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

io_netty.netty_codec_dns.Netty_codec_dnsTest > dnsMessagesManageSectionsFlagsAndReferenceCountedRecords() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > datagramQueryEncoderAndDecoderRoundTripEnvelopeHeaderAndSections() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag() FAILED

io_netty.netty_codec_dns.Netty_codec_dnsTest > registryTypesCodesAndOpcodesResolveKnownAndUnknownValues() SUCCESSFUL

io_netty.netty_codec_dns.Netty_codec_dnsTest > recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords() FAILED


Failures (6):
  JUnit Jupiter:Netty_codec_dnsTest:dnsMessagesManageSectionsFlagsAndReferenceCountedRecords()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'dnsMessagesManageSectionsFlagsAndReferenceCountedRecords', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       io.netty.buffer.Unpooled.<clinit>(Unpooled.java:75)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.ipv4Record(Netty_codec_dnsTest.java:358)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.dnsMessagesManageSectionsFlagsAndReferenceCountedRecords(Netty_codec_dnsTest.java:54)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
     Caused by: java.lang.IllegalArgumentException: Can't find '[toLeakAwareBuffer]' in io.netty.buffer.AbstractByteBufAllocator
       io.netty.util.ResourceLeakDetector.addExclusions(ResourceLeakDetector.java:596)
       io.netty.buffer.AbstractByteBufAllocator.<clinit>(AbstractByteBufAllocator.java:37)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_dns.Netty_codec_dnsTest.recordDecoderExpandsCompressedRecordNamesAndCanonicalNameData(Netty_codec_dnsTest.java:164)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_dns.Netty_codec_dnsTest.ipv4Record(Netty_codec_dnsTest.java:358)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.tcpResponseEncoderAndDecoderRoundTripLengthPrefixFlagsAndAnswers(Netty_codec_dnsTest.java:290)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:datagramQueryEncoderAndDecoderRoundTripEnvelopeHeaderAndSections()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'datagramQueryEncoderAndDecoderRoundTripEnvelopeHeaderAndSections', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.AbstractByteBufAllocator
       io.netty.buffer.ByteBufUtil.<clinit>(ByteBufUtil.java:86)
       io.netty.buffer.ByteBufAllocator.<clinit>(ByteBufAllocator.java:24)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.datagramQueryEncoderAndDecoderRoundTripEnvelopeHeaderAndSections(Netty_codec_dnsTest.java:209)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_dns.Netty_codec_dnsTest.datagramResponseEncoderAndDecoderRoundTripEnvelopeResponseCodeAndTruncatedFlag(Netty_codec_dnsTest.java:249)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
  JUnit Jupiter:Netty_codec_dnsTest:recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords()
    MethodSource [className = 'io_netty.netty_codec_dns.Netty_codec_dnsTest', methodName = 'recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_dns.Netty_codec_dnsTest.recordEncoderAndDecoderRoundTripQuestionsRawRecordsAndDecodePtrRecords(Netty_codec_dnsTest.java:123)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
       [...]

Test run finished after 6 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         7 tests found           ]
[         0 tests skipped         ]
[         7 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::getLong has been called by io.netty.util.internal.PlatformDependent0$4 (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-dns_tests-tests)
WARNING: Please consider reporting this to the maintainers of class io.netty.util.internal.PlatformDependent0$4
WARNING: sun.misc.Unsafe::getLong will be removed in a future release
[         6 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-dns/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-dns_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.


BUILD FAILED in 1m 25s
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.netty:netty-codec-dns:4.1.74.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 26s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.79.Final failed with exit code 1.
FAILED[native-image run][4.1.79.Final][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-dns:4.1.74.Final"]
```

--- a/metadata/io.netty/netty-codec-dns/4.1.79.Final/reachability-metadata.json
+++ b/metadata/io.netty/netty-codec-dns/4.1.79.Final/reachability-metadata.json
@@ -2,6 +2,39 @@
   "reflection": [
     {
       "condition": {
+        "typeReached": "io.netty.handler.codec.dns.TcpDnsResponseEncoder"
+      },
+      "type": "io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueConsumerIndexField",
+      "fields": [
+        {
+          "name": "consumerIndex"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.handler.codec.dns.TcpDnsResponseEncoder"
+      },
+      "type": "io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerIndexField",
+      "fields": [
+        {
+          "name": "producerIndex"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "io.netty.handler.codec.dns.TcpDnsResponseEncoder"
+      },
+      "type": "io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerLimitField",
+      "fields": [
+        {
+          "name": "producerLimit"
+        }
+      ]
+    },
+    {
+      "condition": {
         "typeReached": "io.netty.buffer.AbstractByteBufAllocator"
       },
       "type": "io.netty.buffer.AbstractByteBufAllocator",

